src/main/java/com/redhat/devtools/intellij/tektoncd/ui/bundle/CreateBundleDialog.java

```diff
@@ -0,0 +1,305 @@
+import javax.swing.BoxLayout;
+import javax.swing.Icon;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import javax.swing.JTree;
+import javax.swing.SwingConstants;
+import javax.swing.event.DocumentEvent;
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.TreeSelectionModel;
+import java.awt.BorderLayout;
+import java.awt.Cursor;
+import java.awt.Dimension;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CreateBundleDialog extends BundleDialog {
+    private static final Logger logger = LoggerFactory.getLogger(CreateBundleDialog.class);
+    private Tree tree;
+    private JBList<Resource> layers;
+    private Bundle bundle;
+    private JPanel bundleBodyPanel;
+    private JButton moveToBundle, leaveFromBundle;
+    private JTextField txtValueParam;
+    private MoveToBundleAction moveToBundleAction;
+    private RemoveFromBundleAction removeFromBundleAction;
+
+    public CreateBundleDialog(@Nullable Project project, Tkn tkn) {
+        super(project, "Create and deploy new bundle", "Deploy", tkn);
+    }
+
+
+    @Override
+    protected void preInit() {
+        bundle = new Bundle();;
+    }
+
+    @Override
+    protected String getTopDescriptionText() {
+        return "Publish a new Tekton Bundle to a registry by passing in a set (max 10) of Tekton objects";
+    }
+
+    @Override
+    protected String getTopDescriptionTooltip() {
+        return "Look for the Tekton resources you want to include in your bundle and move them from the tree to the bundle body. " +
+                "Once done, add the image name and push it to the registry.";
+    }
+
+    @Override
+    protected JComponent buildRightPanel() {
+        JPanel panel = new JPanel(new BorderLayout());
+        panel.add(buildBundleBodyPanel(), BorderLayout.CENTER);
+        panel.add(buildSwitchButtonsPanel(), BorderLayout.WEST);
+        return panel;
+    }
+
+    private JComponent buildSwitchButtonsPanel() {
+        initActions();
+
+        moveToBundle = createButton(moveToBundleAction,
+                AllIcons.Actions.ArrowExpand);
+        leaveFromBundle = createButton(removeFromBundleAction,
+                AllIcons.Actions.ArrowCollapse);
+
+        JPanel switchPanel = new JPanel();
+        switchPanel.setLayout(new BoxLayout(switchPanel, BoxLayout.Y_AXIS));
+        switchPanel.add(moveToBundle);
+        switchPanel.add(leaveFromBundle);
+        return switchPanel;
+    }
+
+    private void initActions() {
+        moveToBundleAction = new MoveToBundleAction(bundle,
+                () -> showWarning("The bundle cannot contain more than 10 Tekton objects. " +
```

Comment:
Idem Tekton objects -> Tekton resources. Also layer is probably cryptic for users so I would replace "Please remove some layer before to add new ones" by "Please remove one or several Tekton resources from the bundle before adding new ones"